1Password/shell-plugins
# Add support for Vultr

Add support for Vultr. Fixes https://github.com/1Password/shell-plugins/issues/158.

- The shell plugin supports importing of API Key from `~/.vultr-cli.yaml` file.
- Provisions the credential as an environment variable `VULTR_API_KEY`

## Testing steps

- Sign up for a Vultr account. 
- Get an API Token from [this page](https://my.vultr.com/settings/#settingsapi). You'll need to set up billing to get a token though.
- Set up Vultr CLI using `vultr-cli account`, and create a file at `~/.vultr-cli.yaml` with the content: `api-key: key-goes-here`
- Checkout this branch and run `make vultr/build`
- Run `op plugin init vultr-cli`, and source shell plugin alias as well by running `source ~/.op/plugins.sh`
- Follow on-screen instructions to import your token from the config file.
- Restart shell and run `vultr-cli` and notice that the credential is now fetched from the 1Password item and provisioned as an environment variable.

## Example secret:

```
➜  shell-plugins git:(add/vultr) make vultr/example-secrets
go run cmd/contrib/main.go vultr/example-secrets
API Key:
  API Key: MH0SBC3BVWH27A00Q9CCCVKJKDUWCEXAMPLE
```

Thanks and feedback welcome!
